src/lib/Liblog/setup_env.c

```diff
@@ -157,3 +157,30 @@ setup_env(char *filen)
 	fclose(efile);
 	return (-1);
 }
+
+/**
+ * @brief
+ * 	log_supported_auth_methods() - print PBS_SUPPORTED_AUTH_METHODS to the log file.
+ *
+ * @param[in]	supported_auth_methods = An array of supported methods
+ *
+ * @return	void
+ */
+void
+log_supported_auth_methods(char **supported_auth_methods) {
+
+	if (supported_auth_methods != NULL) {
+		int i;
+		char temp_buf[100 + 1] = {'\0'}; //MAXAUTHNAME 100
```

Comment:
This is wrong, MAXAUTHNAME is max length of individual auth method name length, not max length of the value of `PBS_SUPPORTED_AUTH_METHODS`, that means we don't have a length limit on `PBS_SUPPORTED_AUTH_METHODS` so you can't use a buffer for it.

So to solve the above issue and making code better I have an alternate approach to below for loop:
How about we define one method in libutil called, `join_str_list(list, sep)` (aka reverse method of `break_comm_list()` and use it here:
```c
char *join_str_list(char **list, char *sep)
{
	size_t sep_len = 0;
	int i = 0;
	char *ret = NULL;
	size_t ret_len = 0;

	if (list == NULL || sep == NULL) // Add more check if needed
		return NULL;
	
	sep_len = strlen(sep);
	while(list[i] != NULL) {
		ret_len += strlen(list[i]);
		ret_len += sep_len;
		i++;
	}
	ret = (char *)calloc(1, ret_len + 1);
	if (ret == NULL)
		return NULL;
	i = 0;
	while(list[i] != NULL) {
		strcat(ret, list[i]);
		strcat(ret, sep);
	}
	return ret; // caller has to free this
}
```